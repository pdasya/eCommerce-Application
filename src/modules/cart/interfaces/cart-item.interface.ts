export interface ICartItem {
  id: string;
  sku: string;
  name: string;
  imageUrl: string;
  quantity: number;
  initialPrice: number;
  finalPrice: number;
}
