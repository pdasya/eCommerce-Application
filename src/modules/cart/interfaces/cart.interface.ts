import { ICartItem } from './cart-item.interface';

export interface ICart {
  id: string;
  version: number;
  items: ICartItem[];
  currency: string;
  initialPrice: number;
  discount: number;
  finalPrice: number;
}
