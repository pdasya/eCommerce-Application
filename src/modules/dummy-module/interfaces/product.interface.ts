export interface IProduct {
  id: number;
  name: string;
  price: number;
  size: 's' | 'm' | 'l';
}
