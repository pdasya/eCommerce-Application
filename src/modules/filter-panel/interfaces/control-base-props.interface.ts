import { ForwardedRef } from 'react';
import { IControlHandle } from './control-handle.interface';

export interface IControlBaseProps<T> {
  name: string;
  ref?: ForwardedRef<IControlHandle>;
  onChange?: (payload: [string, T]) => void;
  onDefaultChange?: (isDefault: boolean) => void;
}
