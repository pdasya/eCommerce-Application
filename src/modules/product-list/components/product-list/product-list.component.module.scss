.root {
  display: grid;
  width: fit-content;
  height: 100%;
  gap: 5px;
  grid-template-columns: repeat(5, 1fr);

  @media screen and (width <= 1500px) {
    grid-template-columns: repeat(4, 1fr);
  }

  @media screen and (width <= 1200px) {
    grid-template-columns: repeat(3, 1fr);
  }

  @media screen and (width <= 900px) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media screen and (width <= 600px) {
    grid-template-columns: repeat(1, 1fr);
  }
}
