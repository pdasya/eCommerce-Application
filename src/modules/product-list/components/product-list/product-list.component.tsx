import React, { FC } from 'react';
import { useAppSelector } from '@hooks/use-app-selector.hook';
import { selectProducts } from '@store/catalog/catalog.slice';
import { selectLoading } from '@store/misc/misc.slice';
import { Paper } from '@mui/material';
import styles from './product-list.component.module.scss';
import { ProductCard } from '../product-card/product-card.component';
import { NotFoundBanner } from '../not-found-banner/not-found-banner.component';
import { LoadingBanner } from '../loading-banner/loading-banner.component';
import { ProductSort } from '../product-sorting/product-sorting.component';

export const ProductList: FC = () => {
  const products = useAppSelector(selectProducts);
  const isLoading = useAppSelector(selectLoading);

  return (
    <>
      <Paper className={styles.panel}>
        <ProductSort />
      </Paper>
      <div className={styles.root}>
        {products.length ? (
          products.map(product => <ProductCard {...product} key={product.id} />)
        ) : isLoading ? (
          <LoadingBanner />
        ) : (
          <NotFoundBanner />
        )}
      </div>
    </>
  );
};
