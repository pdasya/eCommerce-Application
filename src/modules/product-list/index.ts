export { ProductList } from './components/product-list/product-list.component';
