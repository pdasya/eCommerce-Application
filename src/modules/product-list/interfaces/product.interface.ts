export interface IProduct {
  id: number;
  title: string;
  description: string;
  imageSrc: string;
  price: number;
  currency: string;
  discountPrice?: number;
}
