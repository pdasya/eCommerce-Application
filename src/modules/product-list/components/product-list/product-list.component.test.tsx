import React from 'react';
import { act } from '@testing-library/react';
import '@testing-library/jest-dom';
import { catalogUpdating, update } from '@store/catalog/catalog.slice';
import { renderWithProviders } from '@/utils/render-with-providers.test-util';
import { ProductList } from './product-list.component';
import { IProduct } from '@/interfaces/interfaces';

describe('Product list', () => {
  test('snapshot match', () => {
    const mockProducts: IProduct[] = [
      {
        currency: 'currency1',
        currentPrice: 'price1',
        description: 'description1',
        id: 'id1',
        imageAlt: 'alt1',
        imageSrc: 'src1',
        slug: 'slug1',
        title: 'title1',
      },
      {
        currency: 'currency2',
        currentPrice: 'price2',
        description: 'description2',
        id: 'id2',
        imageAlt: 'alt2',
        imageSrc: 'src2',
        slug: 'slug2',
        title: 'title2',
        discountPrice: 'description2',
      },
    ];
    const { store, asFragment } = renderWithProviders(<ProductList />);

    expect(asFragment()).toMatchSnapshot('product-list-snapshot-1');

    act(() => store.dispatch(catalogUpdating()));

    expect(asFragment()).toMatchSnapshot('product-list-snapshot-2');

    act(() => store.dispatch(update(mockProducts)));

    expect(asFragment()).toMatchSnapshot('product-list-snapshot-3');
  });
});
