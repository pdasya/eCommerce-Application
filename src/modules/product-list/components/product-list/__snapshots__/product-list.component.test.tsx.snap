// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product list snapshot match: product-list-snapshot-1 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1>
        Loading 
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="HourglassEmptyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 2v6h.01L6 8.01 10 12l-4 4 .01.01H6V22h12v-5.99h-.01L18 16l-4-4 4-3.99-.01-.01H18V2zm10 14.5V20H8v-3.5l4-4zm-4-5-4-4V4h8v3.5z"
          />
        </svg>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Product list snapshot match: product-list-snapshot-2 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1>
        Loading 
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="HourglassEmptyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 2v6h.01L6 8.01 10 12l-4 4 .01.01H6V22h12v-5.99h-.01L18 16l-4-4 4-3.99-.01-.01H18V2zm10 14.5V20H8v-3.5l4-4zm-4-5-4-4V4h8v3.5z"
          />
        </svg>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Product list snapshot match: product-list-snapshot-3 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1>
        Loading 
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="HourglassEmptyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 2v6h.01L6 8.01 10 12l-4 4 .01.01H6V22h12v-5.99h-.01L18 16l-4-4 4-3.99-.01-.01H18V2zm10 14.5V20H8v-3.5l4-4zm-4-5-4-4V4h8v3.5z"
          />
        </svg>
      </h1>
    </div>
  </div>
</DocumentFragment>
`;
